scripts/update_version.py

```
@@ -0,0 +1,582 @@
+                    continue
+                if line.strip() == '#':
+                    continue
+                changelog_lines.append(line)
+        
+        # Remove trailing empty lines
+        while changelog_lines and not changelog_lines[-1].strip():
+            changelog_lines.pop()
+        
+        return '\n'.join(changelog_lines).strip()
+    
+    def _clean_changelog_sections(self, changelog: str) -> str:
```

Ruff (0.11.9)

`_clean_changelog_sections` is too complex (11 > 10)

(C901)